lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/impl/ThreadPoolStage.java

```diff
@@ -206,14 +208,29 @@ public void run() {
    * Closes resources.
    */
   @Override
-  public void close() throws Exception {
+  public void close() {
+
     if (closed.compareAndSet(false, true) && numThreads > 0) {
+
+      LOG.log(Level.FINEST, "Closing ThreadPoolStage {0}: begin", this.name);
+
       executor.shutdown();
-      if (!executor.awaitTermination(shutdownTimeout, TimeUnit.MILLISECONDS)) {
-        LOG.log(Level.WARNING, "Executor did not terminate in " + shutdownTimeout + "ms.");
+
+      boolean isTerminated = false;
+      try {
+        isTerminated = executor.awaitTermination(SHUTDOWN_TIMEOUT, TimeUnit.MILLISECONDS);
+      } catch (final InterruptedException ex) {
+        LOG.log(Level.WARNING, "Interrupted closing ThreadPoolStage " + this.name, ex);
+      }
+
+      if (!isTerminated) {
         final List<Runnable> droppedRunnables = executor.shutdownNow();
-        LOG.log(Level.WARNING, "Executor dropped " + droppedRunnables.size() + " tasks.");
+        LOG.log(Level.WARNING,
+            "Closing ThreadPoolStage {0}: Executor did not terminate in {1} ms. Dropping {2} tasks",
+            new Object[] {this.name, SHUTDOWN_TIMEOUT, droppedRunnables.size()});
       }
+
```

Comment:
In that case, we really should check whether we succeeded and put a stern WARNING into the log when we did not.